### PR TITLE
Enable configuration-cache

### PR DESCRIPTION
**Pull request by Goooler**
https://docs.gradle.org/7.4/userguide/configuration_cache.html

https://blog.gradle.org/introducing-configuration-caching

### CODE REVIEW

**Comment by Goooler** [comment]
```
* What went wrong:
Configuration cache state could not be cached: input property '$3' of ':app:transformClassesWithHiddenApiRefineForRelease': error writing value of type 'org.gradle.api.internal.file.FilteredFileTree'
> Querying the mapped value of provider(interface java.util.Set) before task ':app:compileReleaseKotlin' has completed is not supported
```

File: build.gradle.kts [hunk 1]
```diff
@@ -10,6 +10,18 @@ plugins {
 
 allprojects {
   apply(plugin = rootProject.libs.plugins.kotlinter.get().pluginId)
+
+  tasks.register(":app:transformClassesWithHiddenApiRefineForDebug") {
+    doLast {
+      println(project.file("test.txt"))
+    }
+  }
```
Comment (Goooler) [inline build.gradle.kts]:
```
Could not create task ':app::app:transformClassesWithHiddenApiRefineForDebug'.
> The task name ':app:transformClassesWithHiddenApiRefineForDebug' must not contain any of the following characters: [/, \, :, <, >, ", ?, *, |].

```

File: build.gradle.kts [hunk 2]
```diff
@@ -10,6 +10,12 @@ plugins {
 
 allprojects {
   apply(plugin = rootProject.libs.plugins.kotlinter.get().pluginId)
+
+  tasks.matching {
+    it.name.contains("transformClassesWithHiddenApiRefine")
+  }.configureEach {
+    notCompatibleWithConfigurationCache("transformClassesWithHiddenApiRefine task can't support CC")
+  }
```
Comment (Goooler) [inline build.gradle.kts]:
```
Configuration cache state could not be cached: input property '$3' of ':app:transformClassesWithHiddenApiRefineForRelease': error writing value of type 'org.gradle.api.internal.file.FilteredFileTree'
> Querying the mapped value of provider(interface java.util.Set) before task ':app:compileReleaseKotlin' has completed is not supported
```

**Comment by zhaobozhen** [comment]
rebase 再试一下

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,5 @@
-#Wed Feb 09 19:08:28 CST 2022
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
 distributionPath=wrapper/dists
-zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-rc-1-bin.zip
```
Comment (Goooler) [inline gradle/wrapper/gradle-wrapper.properties]: https://docs.gradle.org/7.5-rc-1/release-notes.html#configuration-cache-improvements

File: build.gradle.kts [hunk 4]
```diff
@@ -10,6 +10,17 @@ plugins {
...
+  tasks.matching {
+    it.name.contains("transformClassesWithHiddenApiRefine")
+  }.configureEach {
+    notCompatibleWithConfigurationCache("https://github.com/RikkaApps/HiddenApiRefinePlugin/issues/9")
+  }
+  tasks.matching {
+    it.name.contains("optimizeReleaseRes")
+  }.configureEach {
+    notCompatibleWithConfigurationCache("Cannot serialize Gradle script object references as these are not supported with the configuration cache.")
+  }
```
Comment (Goooler) [inline build.gradle.kts:23]: https://github.com/zhaobozhen/LibChecker/runs/6416195433?check_suite_focus=true

**Pull request merged**